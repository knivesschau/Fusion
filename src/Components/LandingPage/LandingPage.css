/* mobile styles first, min 320px */
.Landing_Page {
    text-align: center;
}

.Intro_Tagline, .Second_Tagline, .Third_Tagline {
    margin-left: auto;
    margin-right: auto; 
    margin: 1.5em;
    margin-bottom: 1.75em;
    margin-top: -.5em;
    background-color: rgb(190, 190, 190, .9);
    border: 1px solid grey;
    border-radius: 2em; 
    box-shadow: inset -3px -2px 5px 0;
    padding: .75em;
}

#app-title {
    margin-top: .25em;
    color: white; 
    text-shadow: 6px 4px 5px black;
    font-size: 3rem;
    text-transform: uppercase;
}

#title-tag {
    margin-top: -1.75em;
    padding: .25em;
    color: #F9D1B1;
    background-color: rgb(60, 60, 62, .10);
    border: 1px solid transparent;
    border-radius: 5em;
    text-shadow: 5px 3px 5px black;
    font-size: 1.5rem;
}

#pitch1, #pitch2, #pitch3 {
    font-size: 1.40rem;
    color: white;
    text-shadow: 3px 2px 2px black
}

#pitch1, #pitch3 {
    margin-top: .10em;
}

#pitch2 {
    margin-top: -.5em;
}

#tagline1, #tagline2, #tagline3 {
    font-size: 1.10rem;
    font-weight: 500;
    margin-top: -.5em;
    margin-left: .25em;
    margin-right: .25em;
}

#chef-icon, #cookbook-icon, #oven-mitt-icon {
    margin-top: -.25em;
    height: 45%;
    width: 45%;
}

/* tablet and desktop styles */
@media only screen and (min-width: 768px) {
    .Intro_Tagline, .Second_Tagline, .Third_Tagline {
        margin-left: 8em;
        margin-right: 8em;
        border-radius: 5em;
        margin-bottom: 2.5em;
    }

    #app-title {
        font-size: 4rem;
    }

    #title-tag {
        font-size: 2rem;
    }

    #pitch1, #pitch2, #pitch3 {
        font-size: 2rem;
    }

    #tagline1, #tagline2, #tagline3 {
        font-size: 1.35rem;
    }
    
    #chef-icon, #cookbook-icon, #oven-mitt-icon {
        height: 25%;
        width: 25%;
    }
}

@media only screen and (min-width: 1200px) {
    .Intro_Tagline, .Second_Tagline, .Third_Tagline {
        margin-left: 20em;
        margin-right: 20em;
    }

    #app-title {
        font-size: 6rem;
    }

    #title-tag {
        font-size: 3rem;
    }

    #pitch1, #pitch2, #pitch3 {
        font-size: 2.5rem;
        margin-top: .25em;
    }

    #tagline1, #tagline2, #tagline3 {
        font-size: 1.65rem;
        margin-top: -1em;
    }
}

@media only screen and (min-width: 1600px) {
    .Intro_Tagline, .Second_Tagline, .Third_Tagline {
        margin-left: 35em;
        margin-right: 35em;
    }

    #pitch1, #pitch2, #pitch3 {
        margin-top: -.25em;
    }
}
