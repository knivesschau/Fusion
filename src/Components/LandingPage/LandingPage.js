import React, {Component} from 'react';
import chef from '../../images/chef.png';
import cookbook from '../../images/cookbook.png';
import ovenmitt from '../../images/oven-mitt.png';
import './LandingPage.css';

export default class LandingPage extends Component {
    render() {
        return (
            <section className="Landing_Page">

                <header role="banner">
                    <h1 id="app-title">Fusion</h1>
                    <h2 id="title-tag">Let Your Inner Chef Out!</h2>
                </header>

                <div className="Intro_Tagline">
                    <img src={chef} id="chef-icon" alt="chef"/>
                    <h3 id="pitch1">Let's Get Cooking!</h3>

                    <p id="tagline1">
                        Let your inner chef out! Fusion is an online recipe book tailor-made for chefs at all levels, 
                        allowing them to create and save recipe modifications, substitutions, and tweaks to their favorite culinary dishes.
                    </p>
                </div>

                <div className="Second_Tagline">
                    <img src={cookbook} id="cookbook-icon" alt="cookbook"/>
                    <h3 id="pitch2">Fusion Made Easy</h3>

                    <p id="tagline2">
                        Want to adjust the measurements of a dish or use ingredients from another cooking style? 
                        Fusion makes it easy! Fuse and experiment with 14 starter recipes across 7 delicious cuisines and save your 
                        modifications to your personal recipe book. Who knew cooking could be so much fun?
                    </p>
                </div>

                <div className="Third_Tagline">
                    <img src={ovenmitt} id="oven-mitt-icon" alt="oven-mitt"/>
                    <h3 id="pitch3">Chef on the Go</h3>

                    <p id="tagline3">
                        Whether you're on your phone or computer, Fusion allows you to view all of your culinary creations anytime, anywhere. 
                        Fuse, modify, and edit your recipes from your test kitchen or at your desk.
                    </p>
                </div>

            </section>
        );
    };
};