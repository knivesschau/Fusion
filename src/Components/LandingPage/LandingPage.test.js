import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import LandingPage from './LandingPage';

describe ('LandingPage Component', () => {
    it ('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LandingPage/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('Renders the complete LandingPage Component and all nested Components', () => {
        const wrapper = shallow(<LandingPage/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});