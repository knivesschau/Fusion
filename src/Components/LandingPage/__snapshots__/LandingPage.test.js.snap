// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component Renders the complete LandingPage Component and all nested Components 1`] = `
<section
  className="Landing_Page"
>
  <header
    role="banner"
  >
    <h1
      id="app-title"
    >
      Fusion
    </h1>
    <h2
      id="title-tag"
    >
      Let Your Inner Chef Out!
    </h2>
    <h3>
      Note: This version is static and its main purpose is to gather feedback about the basic design, flow and concept. At this time, some buttons and links will not be fully interactive.
    </h3>
  </header>
  <div
    className="Intro_Tagline"
  >
    <h3
      id="pitch1"
    >
      Let's Get Cooking!
    </h3>
    <p
      id="tagline1"
    >
      Let your inner chef out! Fusion is an online recipe book tailor-made for chefs at all levels, allowing them to create and save recipe modifications, substitutions, and tweaks to their favorite culinary dishes.
    </p>
  </div>
  <div
    className="Second_Tagline"
  >
    <h3
      id="pitch2"
    >
      Fusion Made Easy
    </h3>
    <p
      id="tagline2"
    >
      Want to adjust the measurements of a dish or use ingredients from another cooking style? Fusion makes it easy! Fuse and experiment with 10 starter recipes across 5 delicious cuisines and save your modifications to your personal recipe book. Who knew cooking could be so much fun?
    </p>
  </div>
  <div
    className="Third_Tagline"
  >
    <h3
      id="pitch3"
    >
      Chef on the Go
    </h3>
    <p
      id="tagline3"
    >
      Whether you're on your phone or computer, Fusion allows you to view all of your culinary creations anytime, anywhere. Fuse, modify, and edit your recipes from your test kitchen or at your desk.
    </p>
  </div>
</section>
`;
